http-common.c

```diff
@@ -113,36 +124,191 @@ set_method(CURL *ehandle, enum http_method method, struct sized_buffer *body)
       break;
   case HTTP_POST:
       curl_easy_setopt(ehandle, CURLOPT_POST, 1L);
-      //set ptr to payload that will be sent via POST/PUT
-      curl_easy_setopt(ehandle, CURLOPT_POSTFIELDS, body->start);
-      curl_easy_setopt(ehandle, CURLOPT_POSTFIELDSIZE, body->size);
       break;
   case HTTP_PATCH:
       curl_easy_setopt(ehandle, CURLOPT_CUSTOMREQUEST, "PATCH");
-      curl_easy_setopt(ehandle, CURLOPT_POSTFIELDS, body->start);
-      curl_easy_setopt(ehandle, CURLOPT_POSTFIELDSIZE, body->size);
       break;
   case HTTP_PUT:
       curl_easy_setopt(ehandle, CURLOPT_CUSTOMREQUEST, "PUT");
-      curl_easy_setopt(ehandle, CURLOPT_POSTFIELDS, body->start);
-      curl_easy_setopt(ehandle, CURLOPT_POSTFIELDSIZE, body->size);
       break;
   default:
       ERR("Unknown http method (code: %d)", method);
   }
+  
+  if (request_body && request_body->start) {
+    //set ptr to payload that will be sent via POST/PUT
+    curl_easy_setopt(ehandle, CURLOPT_POSTFIELDS, request_body->start);
+    curl_easy_setopt(ehandle, CURLOPT_POSTFIELDSIZE, request_body->size);
+  }
 }
 
+// @todo this should be completely replaced by set_url2
 void
-set_url(CURL *ehandle, char *base_api_url, char endpoint[])
+set_url(CURL *ehandle, char base_api_url[], char url_route[])
 {
   char base_url[MAX_URL_LEN];
-  int ret = snprintf(base_url, sizeof(base_url), "%s%s", base_api_url, endpoint);
+  int ret = snprintf(base_url, sizeof(base_url), "%s%s", base_api_url, url_route);
   ASSERT_S(ret < (int)sizeof(base_url), "Out of bounds write attempt");
 
   CURLcode ecode = curl_easy_setopt(ehandle, CURLOPT_URL, base_url);
   ASSERT_S(CURLE_OK == ecode, curl_easy_strerror(ecode));
 }
 
+void
+set_url2(CURL *ehandle, char base_api_url[], char endpoint[], va_list *args)
+{
+  //create the url route
+  char url_route[MAX_URL_LEN];
+  int ret = vsnprintf(url_route, sizeof(url_route), endpoint, *args);
+  ASSERT_S(ret < (int)sizeof(url_route), "oob write of url_route");
+
+  char base_url[MAX_URL_LEN];
+  ret = snprintf(base_url, sizeof(base_url), "%s%s", base_api_url, url_route);
+  ASSERT_S(ret < (int)sizeof(base_url), "Out of bounds write attempt");
+
+  CURLcode ecode = curl_easy_setopt(ehandle, CURLOPT_URL, base_url);
+  ASSERT_S(CURLE_OK == ecode, curl_easy_strerror(ecode));
+}
+
+static void
+default_cb(void *data)
+{ 
+  return; 
+  (void)data; 
+}
+
+static enum http_action
+default_success_cb(
```

Comment:
the action is defined internal and used internal, it is not really part of HTTP protocol.  The original name `perform_action` is better.

Reply:
Makes sense. Will make the changes.